remote-workstations/new-agent-vm/Install-PCoIPAgent.sh

```diff
@@ -243,131 +238,326 @@ active_d = ldap_lib(args.address, args.user_name, args.password, args.domain)
+INST_LAST_STEP="initial"
 
-sudo wget --retry-connrefused --tries=3 --waitretry=5 -O /etc/yum.repos.d/pcoip.repo $agent_repo_url
+AGENT_CHANNEL=$(tr '[:upper:]' '[:lower:]' <<<"$AGENT_CHANNEL")
+AGENT_TYPE=$(tr '[:upper:]' '[:lower:]' <<<"$AGENT_TYPE")
 
-exitCode=$?
-if [ $exitCode -ne 0 ]
+if [ "$AGENT_TYPE" != "$AGENT_TYPE_STANDARD" ] && [ "$AGENT_TYPE" != "$AGENT_TYPE_GRAPHICS"  ]
 then
-    echo "failed to add teradici repository."
-    # let's define exit code 100 for this case
-    exit 100
+    echo "unknown agent type $AGENT_TYPE."
+    # let's define exit code 105 for this case
+    exit 105
 fi
 
-# Install the PCoIP Agent
-echo "-->Install the PCoIP $AGENT_TYPE agent"
-for idx in {1..3}
-do
-    sudo yum -y install pcoip-agent-$AGENT_TYPE
-    exitCode=$?
+if [ -f $INST_LOG_FILE ]
+then
+    INST_LAST_STEP=$(tail -1 $INST_LOG_FILE) 
+fi
+
+if [ "$INST_LAST_STEP" == "initial" ]
+then
+    echo "start installing pcoip $AGENT_TYPE agent" | tee $INST_LOG_FILE
+
+    echo "step1 starting" | tee -a $INST_LOG_FILE
+
+    sudo yum install wget
+
+    install_SumoLogic
+
+    join_domain
+
+    EXIT_CODE=$?
+
+    if [ $EXIT_CODE -eq 0 ]
+    then    
+        install_gui
+
+        INST_LAST_STEP="step1 done"
+    else
+        INST_LAST_STEP="step1 failure: $EXIT_CODE"
+    fi
+
+    echo "$INST_LAST_STEP" | tee -a $INST_LOG_FILE
+fi	
+
+if [ "$INST_LAST_STEP" == "step1 done" ]
+then 
+    echo "step2 starting" | tee -a $INST_LOG_FILE
+
+    install_pcoip_agent
+    
+    EXIT_CODE=$?
     
-    if [ $exitCode -eq 0 ]
+    if [ $EXIT_CODE -eq 0 ]
     then
-        break
+        INST_LAST_STEP="step2 done"
     else
-        #delay 5 seconds
-        sleep 5
-        sudo yum -y remove pcoip-agent-$AGENT_TYPE
-        if [ $idx -eq 3 ]
-        then
-            echo "failed to install pcoip agent."
-            # let's define exit code 101 for this case
-            exit 101
-        fi
-        #delay 5 seconds        
-        sleep 5
+        INST_LAST_STEP="step2 failure: $EXIT_CODE"
     fi
-done
     
+    echo "$INST_LAST_STEP" | tee -a $INST_LOG_FILE
+fi
+
+if [ "$INST_LAST_STEP" == "step2 done" ]
+then 
+    echo "step3 starting" | tee -a $INST_LOG_FILE
 
-# register license code
-echo "-->Register license code"
-for idx in {1..5}
-do
-    pcoip-register-host --registration-code="$REGISTRATION_CODE"
-    pcoip-validate-license    
-    exitCode=$?
+    register_pcoip_license
     
-    if [ $exitCode -eq 0 ]
-    then
-        break
+    EXIT_CODE=$?
+    
+    if [ $EXIT_CODE -eq 0 ]
+    then 
+        install_idle
+
+        INST_LAST_STEP="step3 done"			
     else
-        if [ $idx -eq 5 ]
-        then
-            echo "failed to register pcoip agent license."
-            # let's define exit code 102 for this case
-            exit 102
+        INST_LAST_STEP="step3 failure: $EXIT_CODE"
+    fi
+
+    echo "$INST_LAST_STEP" | tee -a $INST_LOG_FILE
+fi
+
+if [ $AGENT_TYPE == $AGENT_TYPE_GRAPHICS ]
```

Comment:
String comparisons should be in double quotes.

Reply:
using double quote